File: app/supabase_admin.py
"""
Supabase Admin client for server-only auth operations (e.g. delete user).

Uses SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY. If either is missing,
get_admin_client() returns None and delete-me endpoint returns 503.
"""

import os
from typing import Optional

def get_admin_client():
    url = os.environ.get("SUPABASE_URL", "")
    key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY", "")
    if not url or not key:
        return None
    from supabase import create_client
    return create_client(url, key)

File: frontend/api/supabase-admin.ts
import { createClient, type SupabaseClient } from '@supabase/supabase-js';

/**
 * Server-only Supabase client with service-role privileges.
 *
 * Never import this from client components — the service-role key
 * must stay on the server.
 */

let _admin: SupabaseClient | null = null;

export function getAdminClient(): SupabaseClient {
  if (_admin) return _admin;

  const url = process.env.NEXT_PUBLIC_SUPABASE_URL ?? process.env.SUPABASE_URL;
  const key = process.env.SUPABASE_SERVICE_ROLE_KEY;

  if (!url || !key) {
    throw new Error(
      'Missing SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY — cannot create admin client',
    );
  }

  _admin = createClient(url, key, {
    auth: { autoRefreshToken: false, persistSession: false },
  });

  return _admin;
}
